go/arrow/bitutil/bitmaps_test.go

```diff
@@ -114,13 +114,27 @@ func TestBitmapWriter(t *testing.T) {
 			// {0b00110110, 0b....1010, ........, ........}
 			assert.Equal(t, []byte{0x36, (0x0A | (fillByte & 0xF0)), fillByte, fillByte}, bitmap)
 		}
+		{
```

Comment:
Maybe could add some assertions around the bytes left return value when calling AppendBBools